### ISSUE TITLE
hyperkit: [ERROR] Found a reference to cluster outside the file (max cluster) from cluster 4.0

### ISSUE THREAD

**Issue by joedborg** [CanonicalLtd/multipass#595]
I'm getting this error when trying to launch an instance on MacOS:

```
$ multipass launch --name banana bionic
Starting banana /[2019-02-06T14:15:56.775] [error] [banana] hyperkit: [ERROR] Found a reference to cluster 274877906949 outside the file (max cluster 5186) from cluster 4.0

[2019-02-06T14:15:56.775] [error] [banana] hyperkit: [ERROR] Mirage block device raised exception: Qcow.Make(Base)(Time).Reference_outside_file(_, _)

[2019-02-06T14:15:56.776] [error] [banana] Mirage-block error: Block device is disconnected

[2019-02-06T14:15:56.776] [error] [banana] Could not open mirage-block device: Invalid argument
```

The launch then seems to hang indefinitely.

**Comment by joedborg** [comment]
```
hyperkit: v0.20180403-17-g3e954c

Homepage: https://github.com/docker/hyperkit
License: BSD
```

**Comment by Saviq** [comment]
Hi @joedborg this sounds like the downloaded image got corrupted. If there's no data that you care about in multipass, you can run:

```
sudo sh /Library/Application\ Support/com.canonical.multipass/uninstall.sh
```

And say "Yes" to removing all the data. If you installed with `brew`, `brew zap multipass` would do the same. Please let us know if reinstalling fixed the issue for you.

**Comment by joedborg** [comment]
Hu @Saviq, thanks for the reply.

It seems I downloaded it via the `.pkg` originally, but removed it using the method you shared and then installed it via `brew cask`.  Unfortunately, I'm still getting this error though.

**Comment by joedborg** [comment]
I've also tried forcing a reinstall of `xhyve`, which hasn't fixed the issue.

**Comment by Saviq** [comment]
@joedborg does `xenial` start fine?

**Comment by joedborg** [comment]
@Saviq yeah, it seems to.  it was freshly downloaded; is there any way to do that from the cli for bionic?

**Comment by Saviq** [comment]
@joedborg that would have been what you did originally - `multipass launch bionic`. And having removed Multipass altogether it would've removed the downloaded image, too... I'll try launching a fresh `bionic` image here.

In the mean time, you can try to find and remove the cached image from `/var/root/Library/Application Support/multipassd`...

**Comment by joedborg** [comment]
@Saviq even when i remove that whole dir, I don't have to wait for the image to be pulled again.  Is there any chance it's being cached somewhere else?

**Comment by Saviq** [comment]
@joedborg that'd be very weird... Please try `brew zap multipass`, that should ensure everything is cleared up, and `brew cask install multipass again`. Sorry for this, we need to think of how to ensure the image is intact.

**Comment by joedborg** [comment]
@Saviq Even when I zap it and reinstall, it's not prompting to dowload the image, just straight to `Starting` and then the error messages.

**Comment by Saviq** [comment]
@joedborg ah, it seems the location for the cached bits changed! Please `brew zap multipass`, remove `/var/root/Library/Caches/multipassd` and start from scratch.

**Comment by joedborg** [comment]
Perfect, that's working @Saviq, thanks for the help.  

**TLDR for anyone with the same issue:**
Remove multipass with
```
brew cask zap multipass
```
or
```
sudo sh /Library/Application\ Support/com.canonical.multipass/uninstall.sh
```
(depending on initial install method).

Then, make sure the old cache is dead
```
sudo rm -rf var/root/Library/Caches/multipassd
```

Finally, reinstall
```
brew cask install multipass
```

**Comment by Saviq** [comment]
FWIW Not sure why all this was required... Both the `uninstall.sh` script and `brew zap` should have removed this directory... :/

Glad you got it working!

**Comment by joedborg** [comment]
@Saviq is it the fact that you need to be sudo to remove `/var/root/Library/Caches/multipassd`?

**Comment by Saviq** [comment]
@joedborg yes, but the `uninstall.sh` script wouldn't run without `sudo`, and I believe `brew zap` asks for `sudo` password as appropriate.
